Update types and add command to regenerate types to makefile, add tests/coverage

alexcasalboni opened the pull request:
This is now up-to-date with the latest WIT files 👌 

Note: I have no idea how to test if this actually runs fine.

Also: the .wasm file is now 11MB (and it becomes 24MB with `enableAot: true`).

CLEMENTINATOR commented:
To test you;d need to setup your edgee to use it as a component and load up the demo and do all 3 events and check if you get a request in return

CLEMENTINATOR reviewed (approved):
tested 
```
EdgeeRequest { method: HttpMethod::Post, url: "https://undefined.amplitude.com/2/httpapi", headers: [("Content-Type", "application/json")], body: "{\"options\":{\"min_id_length\":1},\"events\":[{\"library\":\"Edgee\",\"platform\":\"Web\",\"device_id\":\"e28ec7e6-a074-4865-be8c-2553d62796fa\",\"user_agent\":\"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/132.0.0.0 Safari/537.36\",\"language\":\"en-us\",\"ip\":\"127.0.0.0\",\"session_id\":1738078028}]}" }
```

alexcasalboni commented:
@CLEMENTINATOR @Jer1605 please take another look :) I've updated the sample implementation to be more generic (no Amplitude mentions) and to include some utility functions for explicit `PageData`, `TrackData` and `UserData` types. And I've also added some decent defaults for tests and coverage.

CLEMENTINATOR reviewed (approved):
thank you :)
